lib/validationstream.js

```diff
@@ -0,0 +1,88 @@
+var stream = require('stream');
+var tiletype = require('tiletype');
+var mapnik = require('mapnik');
+var stream = require('stream');
+var uploadLimits = require('mapbox-upload-limits');
+var invalid = require('./invalid');
+
+module.exports = ValidationStream;
+module.exports.validType = validType;
+module.exports.validLength = validLength;
+module.exports.validVectorTile = validVectorTile;
+
+function validType(tile) {
+  return tiletype.type(tile.buffer);
+}
+
+function validLength(tile, limit) {
+  limit = limit || uploadLimits.serialtiles.max_tilesize;
+  return tile.buffer.length <= limit;
+}
+
+function validVectorTile(tile, callback) {
+  var info = mapnik.VectorTile.info(tile.buffer);
+  var error;
+
+  // Right now we are just sending a generic error response.
+  // We can return more info from Node Mapnik if we need
+  // but tests currently check for DeserializationError.
+  if (info.errors) {
+    error = {
+      name: 'ValidityError',
+      message: 'Invalid data'      
+    };
+    return callback(error);
```

Comment:
@jakepruitt do we want to return `callback(invalid('invalid message'))` instead here? This way we are matching validation syntax